sieve/sieve.go

```diff
@@ -62,6 +61,10 @@ func New[K comparable, V any](size int, ttl time.Duration) *Sieve[K, V] {
 		ttl = 0
 	}
 
+	if size <= 0 {
```

Comment:
I would have used a uint for the variable then, abd check for 0, but it would have changed the signature, so it would a breaking changes

Reply:
There's precedent for using ints, since that's what `len` returns  (for which there are other good reasons).

But yes, that's a bit theoretical if we don't want to break the API.

Reply:
```golang
package main

import "fmt"

func main() {
	var a uint
	var b uint
	a = 1
	b = 2
	fmt.Println(a - b) // this returns 18446744073709551615
}
```

Calculating with unsigned integers can result in underflow. 
If `len()` returns `uint`, this might leads unintended huge index calculation.